### PR TITLE
[mobile]Set QUIC network idle timeout to 30s

### PR DESCRIPTION
**Pull request by RenjieTang**
Commit Message: Set QUIC network idle timeout to 30s for Envoy Mobile.
Additional Description: A proto knob was added to modify the QUIC config on idle network timeout. The proto is only set for mobile. 
Risk Level: Low
Testing: unit tests
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: mobile only

### CODE REVIEW

**Review by RyanTheOptimist — DISMISSED** [review]

**Comment by RyanTheOptimist** [comment]
Looks like a proto format fix is needed:

```
diff --git a/api/envoy/config/core/v3/protocol.proto b/api/envoy/config/core/v3/protocol.proto
index e37f7ae..3ecad28 100644
--- a/api/envoy/config/core/v3/protocol.proto
+++ b/api/envoy/config/core/v3/protocol.proto
@@ -105,7 +105,7 @@ message QuicProtocolOptions {
 
   // The duration that a QUIC connection stays idle before it closes itself. If this field is not present, QUIC default will be applied.
   google.protobuf.UInt32Value idle_timeout_seconds = 8
-    [(validate.rules).uint32 = {lte: 600 gte: 1}];
+      [(validate.rules).uint32 = {lte: 600 gte: 1}];
 }
 
 message UpstreamHttpProtocolOptions {

```

**Review by RyanTheOptimist — APPROVED** [review]

File: api/envoy/config/core/v3/protocol.proto [hunk 1]
```diff
@@ -102,6 +102,10 @@ message QuicProtocolOptions {
   // A comma-separated list of strings representing QUIC client connection options defined in
   // `QUICHE <https://github.com/google/quiche/blob/main/quiche/quic/core/crypto/crypto_protocol.h>`_ and to be sent by upstream connections.
   string client_connection_options = 7;
+
+  // The duration that a QUIC connection stays idle before it closes itself. If this field is not present, QUIC default will be applied.
+  google.protobuf.UInt32Value idle_timeout_seconds = 8
```
Comment (lizan) [inline api/envoy/config/core/v3/protocol.proto:107]: Per API style this should be google.protobuf.Duration instead of integer in seconds.

**Review by lizan — CHANGES_REQUESTED** [review]

File: api/envoy/config/core/v3/protocol.proto (hunk 1 above)
Comment (RenjieTang) [reply api/envoy/config/core/v3/protocol.proto]: done

File: api/envoy/config/core/v3/protocol.proto [hunk 2]
```diff
@@ -102,6 +102,12 @@ message QuicProtocolOptions {
   // A comma-separated list of strings representing QUIC client connection options defined in
   // `QUICHE <https://github.com/google/quiche/blob/main/quiche/quic/core/crypto/crypto_protocol.h>`_ and to be sent by upstream connections.
   string client_connection_options = 7;
+
+  // The duration that a QUIC connection stays idle before it closes itself. If this field is not present, QUIC default will be applied.
```
Comment (alyssawilk) [inline api/envoy/config/core/v3/protocol.proto:106]: what is the QUIC default?  is it in the spec?

Comment (alyssawilk) [reply api/envoy/config/core/v3/protocol.proto:106]: Also if we think the default is fine for like internal quic corporate proxy but we think 30s is better for internet client facing I think we should document both here and in H3 docs.

**Review by alyssawilk — COMMENTED** [review]
Thanks for picking this up!  A few drive by comments

**Review by lizan — APPROVED** [review]
/lgtm api

**Pull request merged**